app/controllers/edge_controller.rb

```diff
@@ -124,8 +123,13 @@ def all_hosts
     end
   end
 
-  private
+  def hmac_api_key(pass, salt)
```

Comment:
move this function to concerns

Reply:
maybe in next fixes....

Reply:
please fix it, we are trying to make order in the code